sklearn/metrics/_scorer.py

```diff
@@ -1110,8 +1112,75 @@ def from_scorer(cls, scorer, response_method, thresholds):
         instance._metadata_request = scorer._get_metadata_request()
         return instance
 
+    def _scores_from_predictions(
+        self,
+        y_true,
+        y_score,
+        classes=None,
+        **kwargs,
+    ):
+        """Computes scores per threshold, given continuous response and true labels.
+
+        Parameters
+        ----------
+        y_true : array-like of shape (n_samples,)
+            Ground truth (correct) target labels.
+
+        y_score : array-like of shape (n_samples,)
+            Continuous response scores.
+
+        classes: array-like, default=None
+            Class labels. If `None`, inferred from `y_true`.
+
+        **kwargs : dict
+            Parameters to pass to `self.score_func`.
+
+        Returns
+        -------
+        score_thresholds : ndarray of shape (thresholds,)
+            The scores associated with each threshold.
+
+        thresholds : ndarray of shape (thresholds,)
+            The thresholds used to compute the scores.
+        """
+        # This could also be done in `decision_threshold_curve`, not sure which
+        # is better
+        y_true_unique = cached_unique(y_true)
+        if classes is None:
+            classes = y_true_unique
+        # not sure if this separate error msg needed.
+        # there is the possibility that set(classes) != set(y_true_unique) fails
+        # because `y_true` only contains one class.
+        if len(y_true_unique) == 1:
+            raise ValueError("`y_true` only contains one class label.")
+        if set(classes) != set(y_true_unique):
+            raise ValueError(
+                f"`classes` ({classes}) is not equal to the unique values found in "
+                f"`y_true` ({y_true_unique})."
+            )
```

Comment:
These checks could be done in `decision_threshold_curve` instead, not sure which is better.